src/v/kafka/protocol/types.h

```diff
@@ -136,4 +136,20 @@ concept KafkaApi = requires(T request) {
     { T::min_flexible } -> std::convertible_to<const api_version&>;
 };
 
+/*
+ * Data type of the configuration entry.
+ */
+enum class describe_configs_type : int8_t {
+    unknown = 0,
+    boolean = 1,
+    string = 2,
+    _int = 3,
+    _short = 4,
+    _long = 5,
+    _double = 6,
+    list = 7,
+    _class = 8,
+    password = 9
```

Comment:
for list, check for vector type but also see if we return any lists (vectors)

Do your research for class and password type.

Reply:
If we don't already support a type then my constexpr checks should fail on comiple with them.